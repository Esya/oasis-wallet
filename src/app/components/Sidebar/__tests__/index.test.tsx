import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { Navigation } from '..'
import type { UseTranslationResponse } from 'react-i18next'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    } as UseTranslationResponse<'translation'>
  },
}))

const renderComponent = (store: any) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    </Provider>,
  )
}

describe('<Navigation />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
    jest.resetAllMocks()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it.todo('should be responsive')
})
