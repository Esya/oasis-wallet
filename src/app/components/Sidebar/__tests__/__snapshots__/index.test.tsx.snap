// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match snapshot 1`] = `
.c12 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c12 g {
  fill: inherit;
  stroke: inherit;
}

.c12 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c12 *[stroke*="#"],
.c12 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c12 *[fill-rule],
.c12 *[FILL-RULE],
.c12 *[fill*="#"],
.c12 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: component-sidebar;
  border-inline-end: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 220px;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 0px;
  padding-right: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: url(/logo192.png) no-repeat center center;
  background-size: cover;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 48px;
  width: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: background-oasis-blue;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 4px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 12px;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c6 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c13 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c10:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible) > circle,
.c10:focus:not(:focus-visible) > ellipse,
.c10:focus:not(:focus-visible) > line,
.c10:focus:not(:focus-visible) > path,
.c10:focus:not(:focus-visible) > polygon,
.c10:focus:not(:focus-visible) > polyline,
.c10:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c17 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c17:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus > circle,
.c17:focus > ellipse,
.c17:focus > line,
.c17:focus > path,
.c17:focus > polygon,
.c17:focus > polyline,
.c17:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus::-moz-focus-inner {
  border: 0;
}

.c17:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible) > circle,
.c17:focus:not(:focus-visible) > ellipse,
.c17:focus:not(:focus-visible) > line,
.c17:focus:not(:focus-visible) > path,
.c17:focus:not(:focus-visible) > polygon,
.c17:focus:not(:focus-visible) > polyline,
.c17:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    border-inline-end: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding-top: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    border: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c19 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c20 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="/"
    >
      <div
        class="c2"
      >
        <div
          class="c3 StyledAvatar-sc-1suyamb-1"
        />
        <div
          class="c4"
        />
        <span
          class="c5"
        >
          Oasis Wallet
        </span>
      </div>
    </a>
  </div>
  <div
    class="c6"
  />
  <div
    class="c7"
  >
    <nav
      class="c8"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <div
          class="c9"
        >
          <button
            aria-label="menu.home"
            class="c10"
            data-testid="nav-home"
            type="button"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="Home"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
              <div
                class="c13"
              />
              menu.home
            </div>
          </button>
        </div>
      </a>
      <div
        class="c6"
      />
      <div
        class="c6"
      />
    </nav>
  </div>
  <div
    class="c6"
  />
  <nav
    class="c14"
  >
    <div
      class="c15"
    >
      <button
        aria-label="theme.darkMode"
        class="c10"
        type="button"
      >
        <div
          class="c11"
        >
          <svg
            aria-label="Moon"
            class="c12"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.874 5.008c2.728-1.68 6.604-1.014 8.25.197-2.955.84-5.11 3.267-5.242 6.415-.18 4.28 3.006 6.588 5.24 7.152-1.964 1.343-4.36 1.293-5.235 1.172-3.568-.492-6.902-3.433-7.007-7.711-.106-4.278 2.573-6.35 3.994-7.225z"
              stroke="#000"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <div
            class="c13"
          />
          theme.darkMode
        </div>
      </button>
    </div>
    <div
      class="c6"
    />
    <div
      class="c6"
    />
    <div
      class="c16"
    >
      <button
        aria-label="Open Menu"
        class="c17"
        type="button"
      >
        <div
          class="c18"
        >
          <div
            class="c19"
          >
            <svg
              aria-label="Language"
              class="c12"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 23c6.075 0 11-4.925 11-11S18.075 1 12 1 1 5.925 1 12s4.925 11 11 11zm0 0c3 0 4-5 4-11S15 1 12 1 8 6 8 12s1 11 4 11zM2 16h20M2 8h20"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c20"
          >
            <span
              class="c5"
            >
              Language
            </span>
          </div>
          <div
            class="c19"
          >
            <svg
              aria-label="FormDown"
              class="c12"
              viewBox="0 0 24 24"
            >
              <path
                d="m18 9-6 6-6-6"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </nav>
</div>
`;
